### PR TITLE
resolve issue #621

### PR DESCRIPTION
**Pull request by adrianBasa**
user can comment if all the following are met:
-   user is logged in
- user purchased previously the product

### CODE REVIEW

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs [hunk 1]
```diff
@@ -57,7 +63,35 @@ public async Task<IViewComponentResult> InvokeAsync(long entityId, string entity
...
+                var user = await _workContext.GetCurrentUser();
+                model.LoggedUserName = user.FullName;
+
+                var currentUserOrder = _orderRepository
+                    .Query()
+                    .Where(x => x.CreatedById == user.Id);
+                var userOrderedProduct = currentUserOrder
+                    .Where(x => x.OrderItems.Where(y =>y.ProductId == entityId).Count() > 0);
+
+
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs]: please remove blank lines, here and other places

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs [hunk 2]
```diff
@@ -57,7 +63,35 @@ public async Task<IViewComponentResult> InvokeAsync(long entityId, string entity
...
+                var currentUserOrder = _orderRepository
+                    .Query()
+                    .Where(x => x.CreatedById == user.Id);
+                var userOrderedProduct = currentUserOrder
+                    .Where(x => x.OrderItems.Where(y =>y.ProductId == entityId).Count() > 0);
+
+            
+   
+                 model.CanReview = userOrderedProduct.Any() ? true : false;
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs]: I think CanReview is not a good name, it's too general. How about HasBoughtProduct?

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Views/Review/_ReviewForm.cshtml [hunk 3]
```diff
@@ -27,13 +32,22 @@
...
+            <input asp-for="ReviewerName" readonly class="form-control" value="@Model.LoggedUserName" />
             <span asp-validation-for="ReviewerName" class="text-danger"></span>
         </div>
     </div>
+
+
     <div class="form-group row">
         <div class="offset-md-4 col-md-8">
-            <button id="btn-addreview" type="submit" class="btn btn-primary">@Localizer["Submit review"]</button>
+            <button id="btn-addreview" @(!Model.CanReview ? "disabled: disabled": "" ) type="submit" class="btn btn-primary">@Localizer["Submit review"]</button>
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Views/Review/_ReviewForm.cshtml]: The users might not know why this button is disabled

File: src/SimplCommerce.WebHost/appsettings.json [hunk 4]
```diff
@@ -1,41 +0,0 @@
-﻿{
```
Comment (thiennn) [inline src/SimplCommerce.WebHost/appsettings.json]: We shouldn't commit the change in ConnectionStrings, it's specific to developers. We recommend using secret manager http://docs.simplcommerce.com/en/latest/development-tips/

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs (hunk 2 above)
Comment (ksks88) [reply src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs]: You don't need ternary operator, right?!

`model.HasBoughtProduct= userOrderedProduct.Any();`

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs (hunk 1 above)
Comment (ksks88) [reply src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs]: Also could be wrote something like:

`var userOrderedProduct = currentUserOrder
                    .Where(x => x.OrderItems.Any(y =>y.ProductId == entityId));`

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs [hunk 5]
```diff
@@ -4,18 +4,25 @@
...
+using SimplCommerce.Module.Core.Extensions;
+using SimplCommerce.Module.Orders.Models;
 using SimplCommerce.Module.Reviews.Areas.Reviews.ViewModels;
 using SimplCommerce.Module.Reviews.Models;
 
 namespace SimplCommerce.Module.Reviews.Areas.Reviews.Components
 {
     public class ReviewViewComponent : ViewComponent
-    {
+    {
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs]: Remove the extra space

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs [hunk 6]
```diff
@@ -4,18 +4,25 @@
...
     public class ReviewViewComponent : ViewComponent
-    {
+    {  
         private readonly IRepository<Review> _reviewRepository;
+        private readonly IWorkContext _workContext;
+        private readonly IRepository<Order> _orderRepository;
+       
 
-        public ReviewViewComponent(IRepository<Review> reviewRepository)
+        public ReviewViewComponent(IRepository<Review> reviewRepository, IWorkContext workContext, IRepository<Order> orderRepository )
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs]: Remove the extra space

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs [hunk 7]
```diff
@@ -57,7 +64,29 @@ public async Task<IViewComponentResult> InvokeAsync(long entityId, string entity
...
+                    .Where(x => x.OrderItems.Where(y =>y.ProductId == entityId).Count() > 0);
+
+                 model.HasBoughtProduct = userOrderedProduct.Any() ? true : false;
+
+             }
+            else
+            {
+                model.LoggedUserName = string.Empty;
+                model.HasBoughtProduct = false;
+
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs]: Please remove extra lines unless there is a need

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs [hunk 8]
```diff
@@ -57,7 +64,29 @@ public async Task<IViewComponentResult> InvokeAsync(long entityId, string entity
...
+                    .Where(x => x.CreatedById == user.Id);
+                var userOrderedProduct = currentUserOrder
+                    .Where(x => x.OrderItems.Where(y =>y.ProductId == entityId).Count() > 0);
+
+                 model.HasBoughtProduct = userOrderedProduct.Any() ? true : false;
+
+             }
+            else
+            {
+                model.LoggedUserName = string.Empty;
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs]: I'm not sure why you need to set the logged user yourself?!!

File: src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Views/Review/_ReviewForm.cshtml [hunk 9]
```diff
@@ -1,8 +1,13 @@
 ﻿@model SimplCommerce.Module.Reviews.Areas.Reviews.ViewModels.ReviewForm
+ @if (User.IsInRole("customer") && User.Identity.IsAuthenticated == true)
+    { 
 <form id="form-addreview" asp-area="Reviews" asp-controller="Review" asp-action="AddReview" method="post" role="form">
     <input asp-for="EntityId" type="hidden" />
     <input asp-for="EntityTypeId" type="hidden" />
     <div asp-validation-summary="All" class="text-danger"></div>
+
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Views/Review/_ReviewForm.cshtml]: Again extra lines!!

**Pull request merged**